### PR TITLE
Update httpclient related code

### PR DESCRIPTION
**Pull request by dsteinich**
The deprecated httpclient code we were using does not handle SNI. Updated to the replacement modules for SNI support. This includes a couple of "finally" blocks to release the connection resources.

Based on WQP experience, the IdleConnectionMonitorThread was also added to kill any hanging connections which managed to not be cleaned up correctly.

Not sure why the ENDDAT-184 commit didn't make the last cut, but it looks harmless.

### CODE REVIEW

File: nude/src/main/java/gov/usgs/cida/nude/connector/http/AbstractHttpConnector.java [hunk 1]
```diff
@@ -119,19 +122,26 @@ public boolean isReady() {
...
+					methodEntity = makeGetCall();
 					result = new HttpResultSet(methodEntity, this.getParser());
 				} else {
 					log.error("Source not ready: " + uri);
 				}
 			} catch (Exception e) {
 				log.error("Could not make call", e);
+			} finally {
+				try {
+					EntityUtils.consume(methodEntity);
```
Comment (jiwalker-usgs) [inline nude/src/main/java/gov/usgs/cida/nude/connector/http/AbstractHttpConnector.java]: Does the HttpResultSet consume the entity on creation, or is the consumption deferred?  If it is the latter, will this call interfere with the ResultSet?

File: nude-akka/src/main/java/gov/usgs/cida/nude/plan/ConnectorPlanStep.java [hunk 2]
```diff
@@ -42,7 +42,10 @@
 /**
  * [UNIT-TESTABLE]?
  * @author dmsibley
+ * 
+ * Not used in EnDDaT at this time - perhaps elsewhere? Remove @Deprecated if a use is found.
  */
+@Deprecated
```
Comment (jiwalker-usgs) [inline nude-akka/src/main/java/gov/usgs/cida/nude/plan/ConnectorPlanStep.java]: I feel like this is used elsewhere, but I'll have to check.  The name sounds familiar to NAR work, which was following a pattern perhaps on GCMRC.

Comment (jiwalker-usgs) [inline nude-akka/src/main/java/gov/usgs/cida/nude/plan/ConnectorPlanStep.java]: Quick search didn't turn anything up, this seems reasonable, especially being in the akka bits.

File: nude/src/main/java/gov/usgs/cida/nude/connector/http/AbstractHttpConnector.java (hunk 1 above)
Comment (dsteinich) [inline nude/src/main/java/gov/usgs/cida/nude/connector/http/AbstractHttpConnector.java]: Looks like I am in over my head on that one. We need to make sure the stream is closed in order for the connection to be closed. Help?

File: nude/src/main/java/gov/usgs/cida/nude/connector/http/AbstractHttpConnector.java [hunk 3]
```diff
@@ -161,9 +171,9 @@ protected int makeHeadCall(String uri) throws ClientProtocolException, IOExcepti
...
 		if (null != uri) {
 			log.trace("Sending HEAD: " + uri);
-			HttpResponse resp = null;
-		
-			HttpClient httpClient = httpProvider.getClient();
+			CloseableHttpResponse resp = null;
+
+			CloseableHttpClient httpClient = httpProvider.getClient();
 			HttpUriRequest req = new HttpHead(uri);
 			HttpProvider.generateFirefoxHeaders(req, null);
```
Comment (mbucknell) [inline nude/src/main/java/gov/usgs/cida/nude/connector/http/AbstractHttpConnector.java]: Do we really need to generate specific headers for Firefox? I know that this is not part of the ticket. Just asking a question?

File: nude/src/main/java/gov/usgs/cida/nude/connector/http/AbstractHttpConnector.java (hunk 1 above)
Comment (dsteinich) [inline nude/src/main/java/gov/usgs/cida/nude/connector/http/AbstractHttpConnector.java]: New day, new eyes - the HttpResultSet does have a close method that does an EntityUtils.consume(httpEntity), so the happy path should be covered. Would it make sense to have this consume be part of the catch block or just delete it entirely?

**Pull request merged**